test/spec/EnterSpec.hs

```diff
@@ -215,3 +215,81 @@ spec = do
             stdout output `shouldStartWith` "v18."
             output <- runGarn ["enter", "frontend"] "npm --version\nexit\n" repoDir Nothing
             stdout output `shouldStartWith` "9."
+
+        describe "top-level environments" $ around onTestFailureLogger $ do
+          it "allows entering top-level environments" $ \onTestFailureLog -> do
+            writeFile
+              "garn.ts"
+              [i|
+                import * as garn from "#{repoDir}/ts/mod.ts"
+
+                const testTool = garn.build(`
+                  mkdir -p $out/bin
+                  echo 'echo test tool output' > $out/bin/test-tool
+                  chmod +x $out/bin/test-tool
+                `);
+
+                export const topLevelEnv = garn.emptyEnvironment.withDevTools([testTool]);
+              |]
+            output <- runGarn ["enter", "topLevelEnv"] "test-tool\nexit\n" repoDir Nothing
+            onTestFailureLog output
+            stdout output `shouldBe` "test tool output\n"
+
+          it "does not show a description for top-level environments" $ \onTestFailureLog -> do
+            writeFile
+              "garn.ts"
+              [i|
+                import * as garn from "#{repoDir}/ts/mod.ts"
+
+                export const topLevelEnv = garn.emptyEnvironment;
+              |]
+            output <- runGarn ["enter", "--help"] "" repoDir Nothing
+            onTestFailureLog output
+            stdout output
+              `shouldMatch` unindent
+                [i|
+                  Available commands:
+                    topLevelEnv[ ]*
+                |]
+
+        describe "--help" $ around onTestFailureLogger $ do
+          it "allows adding descriptions to environments" $ \onTestFailureLog -> do
+            writeFile
+              "garn.ts"
+              [i|
+                import * as garn from "#{repoDir}/ts/mod.ts"
+
+                export const topLevelEnv =
+                  garn.emptyEnvironment
+                    .setDescription("my test environment");
+              |]
+            output <- runGarn ["enter", "--help"] "" repoDir Nothing
+            onTestFailureLog output
+            stdout output
+              `shouldMatch` unindent
+                [i|
+                  Available commands:
+                    topLevelEnv[ ]*my test environment
+                |]
+
+          it "does not show (or allow entering) the default environment of projects" $ \onTestFailureLog -> do
```

Comment:
I imagine the point is that you can just enter `p` directly? If so, shouldn't there be a test for that? If not, then why do we want this?

Reply:
The changes in this PR meant that the `defaultEnvironment` of projects would 1. show up in the help as e.g. `p.defaultEnvironment` and 2. you could enter it fully qualified, e.g. with `garn enter p.defaultEnvironment`. I don't think that's useful, since you can just enter the project instead (with `garn enter p`). And we already have tests for that. (For example [here](https://github.com/garnix-io/garn/blob/0e61454d1506bb95a4d3a61eb4791cf82b24bfdc/test/spec/EnterSpec.hs#L29-L48).)

So this test here is only about making sure that 1. `p.defaultEnvironment` doesn't show up in the help output and 2. that `garn enter p.defaultEnvironment` doesn't work.

Maybe the test name should be better to clarify? Or do you still think we should have yet another test for entering the default environment of projects?

Reply:
No, you're right. I miss the existing test!